src/gpb_parse.yrl

```diff
@@ -410,6 +412,9 @@ flatten_fields(FieldsOrDefs, FullName) ->
                             {[F | Fs], Ds};
                        (#gpb_oneof{}=O, {Fs,Ds}) ->
                             {[O | Fs], Ds};
+                       ({{extend, Msg},Exts}, {Fs,Ds}) ->
+                            QExt = {{extend, ['.',Msg]}, Exts},
```

Comment:
Excellent! I think you're right in concern about qualification. With a small change of your clause in `flatten_fields` into this...
```erlang
                       ({{extend, _Msg},_Exts}=Def, {Fs,Ds}) ->
                            QDefs = flatten_qualify_defnames(
                                      [Def], drop_last_level(FullName)),
                            {Fs, QDefs++Ds};
```
...then the following extra test will also pass:
```erlang
parses_nested_extending_msgs_in_package_test() ->
    {ok,Defs} = parse_lines(["package p1.p2;",
                             "message m1 {",
                             "  required uint32 f1=1 [default=17];",
                             "  extensions 200 to 299;",
                             "  extend m1 {",
                             "    optional uint32 f2=2;",
                             "  }",
                             "}"]),
    [{package,'p1.p2'},
     {{extensions,'p1.p2.m1'},[{200,299}]},
     {{msg,'p1.p2.m1'},
      [#?gpb_field{name=f1, fnum=1, rnum=2, opts=[{default,17}],
                   occurrence=required},
       #?gpb_field{name=f2, fnum=2, rnum=3, opts=[],
                   occurrence=optional}]}] =
        do_process_sort_defs(Defs, [use_packages]).
```
I can amend your commit with this, if you want(?) or if you want to update the commit and force-push yourself? Whichever way is fine for me